lib/trento/activity_logging/parser/metadata_enricher.ex

```diff
@@ -79,7 +80,15 @@ defmodule Trento.ActivityLog.Logger.Parser.MetadataEnricher do
   defp detect_enrichment(:cluster, {_, %{cluster_id: id}}), do: {:ok, id}
   defp detect_enrichment(:database, {_, %{database_id: id}}), do: {:ok, id}
   defp detect_enrichment(:sap_system, {_, %{sap_system_id: id}}), do: {:ok, id}
+  defp detect_enrichment(:user, {_, %{user_id: id}}), do: {:ok, id}
 
   defp detect_enrichment(_target_entity, {_activity, _metadata}),
     do: {:error, :no_enrichment_needed}
+
+  defp clean_metadata(%{username: username} = metadata) do
+    clean_username = username |> String.split("__") |> List.first()
+    Map.put(metadata, :username, clean_username)
+  end
+
+  defp clean_metadata(metadata), do: metadata
```

Comment:
What are your thoughts about handling the case with usernames like `foo__bar__2024...`?

Reply:
I didn't get your point sorry :upside_down_face: 

What's the question?

Reply:
With the current implementation, a user with username `foo__bar`, upon deletion, will have the username just `foo` persisted, IIUC. However we want the actual user name `foo__bar` to be used, not the truncation that would occur due to usage of two successive `_`s. The timestamp also uses the same `__` as separator.